Jellyfin.Api/Controllers/ConfigurationController.cs

```diff
@@ -0,0 +1,128 @@
+        {
+            return Ok(_configurationManager.Configuration);
+        }
+
+        /// <summary>
+        /// Updates application configuration.
+        /// </summary>
+        /// <param name="configuration">Configuration.</param>
+        /// <returns>Status.</returns>
+        [HttpPost("Configuration")]
+        [Authenticated(Roles = "Admin")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public ActionResult UpdateConfiguration([FromBody, BindRequired] ServerConfiguration configuration)
+        {
+            _configurationManager.ReplaceConfiguration(configuration);
+            return Ok();
+        }
+
+        /// <summary>
+        /// Gets a named configuration.
+        /// </summary>
+        /// <param name="key">Configuration key.</param>
+        /// <returns>Configuration.</returns>
+        [HttpGet("Configuration/{Key}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public ActionResult<object> GetNamedConfiguration([FromRoute] string key)
+        {
+            return Ok(_configurationManager.GetConfiguration(key));
+        }
+
+        /// <summary>
+        /// Updates named configuration.
+        /// </summary>
+        /// <param name="key">Configuration key.</param>
+        /// <returns>Status.</returns>
+        [HttpPost("Configuration/{Key}")]
+        [Authenticated(Roles = "Admin")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ActionResult> UpdateNamedConfiguration([FromRoute] string key)
+        {
+            var configurationType = _configurationManager.GetConfigurationType(key);
+            /*
+            // TODO switch to System.Text.Json when https://github.com/dotnet/runtime/issues/30255 is fixed.
+            var configuration = await JsonSerializer.DeserializeAsync(Request.Body, configurationType);
+            */
+
+            var configuration = await _jsonSerializer.DeserializeFromStreamAsync(Request.Body, configurationType)
+                .ConfigureAwait(false);
+            _configurationManager.SaveConfiguration(key, configuration);
+            return Ok();
+        }
+
+        /// <summary>
+        /// Gets a default MetadataOptions object.
+        /// </summary>
+        /// <returns>MetadataOptions.</returns>
+        [HttpGet("Configuration/MetadataOptions/Default")]
+        [Authenticated(Roles = "Admin")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public ActionResult<MetadataOptions> GetDefaultMetadataOptions()
+        {
+            return Ok(new MetadataOptions());
+        }
+
+        /// <summary>
+        /// Updates the path to the media encoder.
+        /// </summary>
+        /// <param name="mediaEncoderPath">Media encoder path form body.</param>
+        /// <returns>Status.</returns>
+        [HttpPost("MediaEncoder/Path")]
+        [Authenticated(Roles = "Admin", AllowBeforeStartupWizard = true)]
```

Comment:
Why are you using `Authenticated` instead of `Authorize`?

Reply:
I was using the attribute from Emby.Api. We didn’t have any discussion on authentication so I figured it would be the same for now

Reply:
If we do the migration, it should be 100% if you ask me. I'd like to keep Jellyfin.Api relatively clean. I already laid the foundation for authorization and authentication.